### PR TITLE
Add right-click menu to Run with PowerShell for ps1 scripts

### PR DESCRIPTION
**Pull request by machv**
# PR Summary

## PR Context

Classic PowerShell has an option in the file context menu to easily run a PowerShell script with just a right-click on the PowerShell file. With this change to the MSI setup, the same option would be provided also for PowerShell Core installations when installed using the MSI installer on Windows.

By default this option is not enabled, user can select this during interactive setup:
image

Or using a property (`ADD_EXPLORER_CONTEXT_MENU_OPENPOWERSHELL=1`) in case of unattended installation.

This option would be then added in the right-click context menu for `.ps1` files:
image

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
- **User-facing changes**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.

### CODE REVIEW

**Comment by iSazonov** [comment]
@heaths Could you please review the PR?

File: assets/Product.wxs [hunk 1]
```diff
@@ -246,6 +249,17 @@
               </RegistryKey>
               <RemoveRegistryKey Id="RemoveOldOpenKey" Root="HKCR" Key="Directory\ContextMenus\$(var.ProductName)$(var.SimpleProductVersion)$(sys.BUILDARCH)\shell\open" Action="removeOnInstall"/>
             </Component>
+            <Component Id="PowerShellFileContextMenu">
+              <Condition>ADD_FILE_CONTEXT_MENU_RUNPOWERSHELL</Condition>
+              <!-- When right-clicking ps1 file -->
+              <RegistryKey Root="HKCR" Key="Microsoft.PowerShellScript.1\Shell\$(var.ProductName)$(var.SimpleProductVersion)$(sys.BUILDARCH)">
+                <RegistryValue Type="string" Name="MUIVerb" Value="$(var.PowerShellFileContextSubMenuDialogText)"/>
+                <RegistryValue Type="string" Name="Icon" Value="[VersionFolder]pwsh.exe"/>
```
Comment (heaths) [inline assets/Product.wxs:257]: To avoid auto-picking key path, mark this one `KeyPath="yes"`.

File: assets/Product.wxs [hunk 2]
```diff
@@ -246,6 +249,17 @@
...
+              </RegistryKey>
+              <RegistryKey Root="HKCR" Key="Microsoft.PowerShellScript.1\Shell\$(var.ProductName)$(var.SimpleProductVersion)$(sys.BUILDARCH)\Command">
```
Comment (heaths) [inline assets/Product.wxs:259]: Tip: this isn't wrong, but you can also just use a `RegistryKey` element under the previous one (nested) without a root and just `Key="Command"` to have a hierarchical and less structure and less verbose authoring.

**Review by heaths — APPROVED** [review]

File: assets/Product.wxs (hunk 1 above)
Comment (machv) [reply assets/Product.wxs:257]: I will mark this `Icon` value as KeyPath, but I suppose any specific value element might be marked?

File: assets/Product.wxs (hunk 2 above)
Comment (machv) [reply assets/Product.wxs:259]: That's a great tip, anything that minimizes the number of absolute paths is good, will do.

File: assets/Product.wxs (hunk 1 above)
Comment (heaths) [reply assets/Product.wxs]: I would choose that one that the MUIVerb contains potentially more variability but for the same path to pwsh.exe. You don't want the component generating a different GUID just for a display name while representing the same shared file.

Comment (machv) [reply assets/Product.wxs]: Thank you for the explanation. In commit 16aa665 element with Icon is marked as KeyPath.

**Review by iSazonov — APPROVED** [review]

**Comment by iSazonov** [comment]
@machv Please rebase and update Files.wxs because we moved to new .Net 5 Preview version.

**Comment by iSazonov** [comment]
@TravisEz13 Could you please review the PR? Can we merge?

**Review by TravisEz13 — APPROVED** [review]

**Comment by iSazonov** [comment]
@machv Thanks for your contribution!

**Pull request merged**